### PR TITLE
Merge EstablishClaim::Claim and EndProduct

### PR DESCRIPTION
**Pull request by shanear**
They represent the same thing, so merge them and allow the
object to be used for reads & writes.

connects #1509

### CODE REVIEW

File: app/controllers/establish_claims_controller.rb [hunk 1]
```diff
@@ -96,7 +96,7 @@ def email_params
 
   def establish_claim_params
     params.require(:claim)
-          .permit(:modifier, :end_product_code, :end_product_label, :end_product_modifier, :gulf_war_registry,
```
Comment (shanear) [inline app/controllers/establish_claims_controller.rb]: `modifier` was unused

File: app/models/tasks/establish_claim.rb [hunk 2]
```diff
@@ -217,72 +217,3 @@ def joins_task_result
     end
   end
 end
-
-# Class used for validating the claim object
-class EstablishClaim::Claim
```
Comment (joofsh) [inline app/models/tasks/establish_claim.rb]: 🎉

File: app/models/end_product.rb [hunk 3]
```diff
@@ -98,6 +121,17 @@ def from_bgs_hash(hash)
       )
     end
 
+    def from_establish_claim_params(hash)
```
Comment (joofsh) [inline app/models/end_product.rb]: Do we need this? Could we align `EndProduct` with what is sent from the frontend? Or align the frontend with what EndProduct expects? Seems non-ideal that we would map twice, once from the frontend and then again to vbms. 

Maybe a tech improvement thing for later refactors

Comment (shanear) [inline app/models/end_product.rb]: Agreed. I'm just staying hands off of the front end while you bulldoze it ☺ 

I'll create a tech improvement for it

**Pull request merged**